komga/src/main/kotlin/org/gotson/komga/domain/service/BookLifecycle.kt

```diff
@@ -272,4 +281,23 @@ class BookLifecycle(
       eventPublisher.publishEvent(DomainEvent.ReadProgressDeleted(progress))
     }
   }
+
+  fun deleteBookFiles(book: Book) {
+    if (book.path.notExists() || !book.path.isWritable())
+      throw FileNotFoundException("File is not accessible : ${book.path}").withCode("ERR_1018")
+
+    val thumbnails = thumbnailBookRepository.findAllByBookId(book.id)
+      .filter { it.type == ThumbnailBook.Type.SIDECAR }
```

Comment:
it's best to implement the filtering in the repository method, especially since there is already `findAllByBookIdAndType`